lib/ast/AST.cpp

```diff
@@ -14,6 +36,121 @@ void KOREObjectSymbol::addSort(KOREObjectSort *Sort) {
   sort = Sort;
 }
 
+bool KOREObjectSymbol::operator==(KOREObjectSymbol other) const {
+  if (name != other.name || arguments.size() != other.arguments.size()) {
+    return false;
+  }
+  for (int i = 0; i < arguments.size(); i++) {
+    if (!(*arguments[i] == *other.arguments[i])) return false;
+  }
+  return true;
+}
+
+std::string KOREObjectSymbol::layoutString() const {
+  std::string result;
+  for (auto arg : arguments) {
+    auto sort = dynamic_cast<KOREObjectCompositeSort *>(arg);
+    if (sort->getName() == "Map") {
```

Comment:
Not a big deal for a PoC, but we should be using an enum here instead of strings comparisons, it also allows for using a switch.